js/files.ts

```diff
@@ -29,14 +29,51 @@ export const stdout = new File(1);
 /** An instance of `File` for stderr. */
 export const stderr = new File(2);
 
-// TODO This is just a placeholder - not final API.
-export type OpenMode = "r" | "w" | "w+" | "x";
+export enum OpenMode {
+  /** Read-only, default mode.
+   * Starts at beginning of file.
+   */
+  Read = "r",
+  /** Read-write
+   * Start at beginning of file.
+   */
+  ReadWrite = "r+",
+  /** Write-only
+   * Opens and truncates existing file or creates new one
+   * for writing only.
+   */
+  Write = "w",
+  /** Write-read
+   * Opens and truncates existing file or creates new one
+   * for writing and reading.
+   */
+  WriteRead = "w+",
+  /** Append-only
+   * Opens existing file or creates new one.
+   * Each write appends content to file and doesn't overwrite
+   * already existing content.
+   */
+  Append = "a",
+  /** Append-read
+   * Behaves like `OpenMode.Append` and allows to read from file.
+   */
+  AppendRead = "a+",
+  /** Exclusive create
+   * Creates new file only if one doesn't exist already.
+   * Allows to write to file.
+   */
+  ExclusiveCreate = "x",
+  /** Exclusive create - read
+   * Behaves like `OpenMode.ExclusiveCreate` and allows to read from file.
+   */
+  ExclusiveCreateRead = "x+"
+}
```

Comment:
It would be nice if the JSDocs here were single lines... if possible. It seems like many of them could be.

Reply:
Done